src/main/java/step4/RacingGame.java

```diff
@@ -0,0 +1,99 @@
+package step4;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.sun.deploy.util.StringUtils;
+
+public class RacingGame {
+
+    private int moveCount;
+    private List<Car> carPositions;
+    private List<List<Car>> moveCarPositions;
+    private String winnerNames;
+
+    private static final int MOVE_COUNT_NUMBER = 0;
+    private static final int CAR_POSITIONS_SIZE = 0;
+
+    public RacingGame() {
+    }
+
+    public RacingGame(int moveCount, List<Car> carPositions) {
+        validateMoveCount(moveCount);
+        validateCarPositions(carPositions);
+        this.moveCount = moveCount;
+        this.carPositions = carPositions;
+    }
+
+    public void racingByRepeatCount() {
+        List<List<Car>> moveCarPositions = new ArrayList<>();
+        for (int i = 0; i < moveCount; i++) {
+            carPositions = racing();
+            moveCarPositions.add(carPositions);
+        }
+        this.moveCarPositions = moveCarPositions;
+    }
+
+    public List<String> getWinnerNames(List<Car> winners) {
+        return winners.stream().
+                flatMap(car -> Stream.of(car.getName()))
```

Comment:
```suggestion
                map(Car::getName)
```
flatmap을 별도로 사용한 이유가 따로 있으실까요? 🤔